src/Options/ClientOptions.php

```diff
@@ -61,37 +61,37 @@ class ClientOptions implements ArrayAccess
 {
     use OptionsTrait;
 
-    private ?string $apiEndpoint;
+    public ?string $apiEndpoint;
```

Comment:
This is a highly opinionated change. My believe is that the options objects should be mostly structural and have easy of use for the user and the developers.

If you are in need or wanting to use this object, using an IDE and start typing `$options->` should give a quick auto completion that helps the user understand the different options. We should document these values if we decided to continue with this philosophy.

Due this being highly opinionated though, I am willing to return this to `private`.

Reply:
The only concern I have here is that if there's some logic being done in the setters (which is the case for a few of these), it would be lost in setting these manually, which may lead to some confusing behavior.